tgui/packages/tgui/interfaces/PrisonerManagement.tsx

```diff
@@ -0,0 +1,241 @@
+
+type IDInfo = {
+  name: string;
+  points: number;
+  goal: number;
+};
+
+type DMButton = {
+  name?: string;
+  icon?: string;
+  color?: string;
+  tooltip?: string;
+  action_key: string;
+  action_params: Record<string, string>;
+};
+
+type ImplantInfo = {
+  info: Record<string, string>;
+  buttons: DMButton[];
+  category: string;
+  ref: byondRef;
+};
+
+type Data = {
+  authorized: BooleanLike;
+  inserted_id: IDInfo | null;
+  implants: ImplantInfo[];
+};
+
+const ImplantDisplay = (props: { implant: ImplantInfo }) => {
+  const { act } = useBackend<ImplantInfo>();
+  const { info, buttons, ref } = props.implant;
+
+  return (
+    <Stack vertical>
+      <Stack.Item>
+        <LabeledList>
+          {Object.entries(info).map(([key, value]) => (
+            <LabeledList.Item key={key} label={key}>
+              {value}
+            </LabeledList.Item>
+          ))}
+          {buttons.length !== 0 && (
+            <LabeledList.Item label={'Options'}>
+              {buttons.map((button) => (
+                <Button
+                  key={button.action_key}
+                  icon={button.icon}
+                  color={button.color}
+                  tooltip={button.tooltip}
+                  onClick={() =>
+                    act('handle_implant', {
+                      implant_ref: ref,
+                      implant_action: button.action_key,
+                      ...button.action_params,
+                    })
+                  }>
+                  {button.name}
+                </Button>
+              ))}
+            </LabeledList.Item>
+          )}
+          <LabeledList.Divider />
+        </LabeledList>
+      </Stack.Item>
+    </Stack>
+  );
+};
+
+const AllImplantDisplay = (props: { implants: ImplantInfo[] }) => {
+  const implantsByCategory: Record<string, ImplantInfo[]> =
+    props.implants.reduce((acc, implant) => {
+      if (!acc[implant.category]) {
+        acc[implant.category] = [];
+      }
+      acc[implant.category].push(implant);
+      return acc;
+    }, {});
+
+  const [implantTab, setImplantTab] = useLocalState(
+    'implantTab',
+    Object.keys(implantsByCategory)[0]
+  );
+
+  // Converts a category ("tracking implant") to a more readable format ("Tracking")
+  const formatCategory = (category: string) => {
+    return category
+      .split(' ')
+      .map((word) => word.charAt(0).toUpperCase() + word.slice(1))[0];
+  };
+
+  return (
+    <Stack fill vertical>
+      <Stack.Item>
+        <Tabs>
+          {Object.entries(implantsByCategory).map(([category, implants]) => (
+            <Tabs.Tab
+              key={category}
+              selected={implantTab === category}
+              onClick={() => setImplantTab(category)}>
+              {formatCategory(category)}
+            </Tabs.Tab>
+          ))}
+        </Tabs>
+      </Stack.Item>
+      <Stack.Item>
+        {implantsByCategory[implantTab] &&
+          implantsByCategory[implantTab].map((implant) => (
```

Comment:
```suggestion
        {implantsByCategory[implantTab]?.map((implant) => (
```
OR better is to ensure implantsByCategory is always an array via `sortImplants`